purchase_invoice_new_picking_line/models/purchase_order.py

```diff
@@ -0,0 +1,80 @@
+# SPDX-FileCopyrightText: 2024 Coop IT Easy
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+from odoo import api, fields, models
+from odoo.fields import Command
+from odoo.tools.float_utils import float_compare, float_round
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    def _prepare_account_move_lines_from_stock_move(
+        self, stock_move_id, account_move_id
+    ):
+        self.ensure_one()
+
+        qty = max(stock_move_id.quantity_done, 0)
+        date = stock_move_id.date or fields.Date.today()
+        supplierinfo = self.env["product.supplierinfo"]
+        price_unit = 0.0
+        if self.partner_id:
+            supplierinfo = stock_move_id.product_id._select_seller(
+                partner_id=self.partner_id,
+                quantity=qty,
+                date=date,
+                uom_id=stock_move_id.product_uom,
+                params={"order_id": self},
+            )
+        if supplierinfo:
+            price_unit = supplierinfo.price
+        else:
+            price_unit = stock_move_id.product_id.lst_price
```

Comment:
I am not entirely sure about this. Previously, the `price_unit` came from the identically named field on `stock.move`. This field still exists in 16, but only contains useful data in some corner-cases.

I wrote the quoted snippet under the impression that it should be sufficient. However, upstream in `purchase`, the code to get the unit price for a `purchase.order.line` is much much much more complicated. See [this permalink](https://github.com/odoo/odoo/blob/b5b23c5c2f79678c8de9f1ebf6435246cb86fa23/addons/purchase/models/purchase.py#L1244). There is no way to easily re-use this code except copy-pasting and adjusting it, but I do not frankly understand some of what is going on there.

I am hopeful that the above will suffice. If not, I see two options:

1. Copy-paste the complicated linked code. Not great.
2. Pass the parcel. Instead of creating invoice lines from `stock.move`s, create `purchase.order.line`s from `stock.move`s upon validating `stock.picking`s. This seems doable, but I'm not sure if it's semantically correct, because these products were never _ordered_ as such.